addons/product/models/product_template.py

```diff
@@ -774,6 +788,33 @@ def _get_mapped_attribute_names(self, parent_combination=None):
             for attribute_value in all_product_attribute_values
         }
 
+    def _check_attributes_config(self, combination):
+        """Return whether the config of attributes on the template is correct or not.
+
+        :param combination: the combination to check for possibility
+        :type combination: recordset `product.template.attribute.value`
+
+        :return: whether the config of attributes is correct or not
+        :rtype: bool
+        """
+        self.ensure_one()
+
+        if len(combination) != len(self.valid_product_template_attribute_line_ids):
+            # number of attribute values passed is different than the
+            # configuration of attributes on the template
+            return False
+
+        if self.valid_product_template_attribute_line_ids != combination.attribute_line_id:
+            # combination has different attributes than the ones configured on the template
+            return False
+
+        valid_active_values_ids = self.valid_product_template_attribute_line_ids.product_template_value_ids._only_active().ids
+        if not all([id in valid_active_values_ids for id in combination.ids]):
```

Comment:
Here is the results where it is incorrect (I think):
```py
> self.valid_product_template_attribute_line_ids.product_template_value_ids._only_active()
product.template.attribute.value(2378,)
> combination
product.template.attribute.value(2377,)
> self.valid_product_template_attribute_line_ids.product_template_value_ids._only_active() < combination
False
```

So according to the comment it means that combination has not different values than the ones configured on the template but that is not correct, it is?

So I think the solution is just to add a `not`

Reply:
Ok I understand indeed, the condition should return True in your problematic example so that the method can return False.

This is an issue because of the negation as you mentioned.

The non-negated condition would be to use `>=`, but when working with sets, negating `>=` by using `<` does not work, and this is the mistake.

So the solution is probably to do  "not ... >=".
`&` will not work either because it is intersection, but which contains which is important in this case. Combination must be contained in possible values, not the other way around.

We want:
{1} vs {2} = True (True in condition to return False in method)
{2} vs {1} = True
{1,2} vs {2} = False
{1} vs {1,2} = True

I think the only operator that works for all those is `not(valid >= combination)` (or the same `not(combination <= valid)`)

Reply:
Indeed it works fine! My test was failing when it was not returning `False` so I don't know if the assert on `_is_combination_possible_by_config` is really necessary or not. I added the case for the non_variant too